src/main/java/Storage.java

```diff
@@ -0,0 +1,101 @@
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.time.format.DateTimeFormatter;
+import java.util.ArrayList;
+import java.util.Scanner;
+
+/**
+ * Encapsulates the storage for the application.
+ * Each storage has a file path which reveals
+ * the file path of the text file used to load and save all tasks.
+ */
+public class Storage {
+
+    private String filePath = "";
+
+    public Storage(String filePath) {
+        this.filePath = filePath;
+    }
+
+    /**
+     * Loads the previously curated task list.
+     * @return previously curated task list
+     * @throws DukeException exception specific to Duke application
+     */
+    public ArrayList<Task> load() throws DukeException {
+        ArrayList<Task> outputList = new ArrayList<>();
+        try {
+            File f = new File(filePath);
+            Scanner s = new Scanner(f);
+            while (s.hasNext()) {
+                String savedTask = s.nextLine();
+                String[] savedTaskSplit = savedTask.split("\\|");
+                if (savedTaskSplit[0].equals("T")) {
+                    String description = savedTaskSplit[2];
+                    Task todoTask = new Todo(description);
+                    if (savedTaskSplit[1].equals("1")) {
+                        todoTask.markAsDone();
+                    }
+                    outputList.add(todoTask);
+                } else if (savedTaskSplit[0].equals("E")) {
+                    String description = savedTaskSplit[2];
+                    String at = savedTaskSplit[3];
+                    Task eventTask = new Event(description, at);
+                    if (savedTaskSplit[1].equals("1")) {
+                        eventTask.markAsDone();
+                    }
+                    outputList.add(eventTask);
+                } else {
+                    String description = savedTaskSplit[2];
+                    String by = savedTaskSplit[3];
+                    Task deadlineTask = new Deadline(description, by);
+                    if (savedTaskSplit[1].equals("1")) {
+                        deadlineTask.markAsDone();
+                    }
+                    outputList.add(deadlineTask);
+                }
+            }
+        } catch (FileNotFoundException e) {
```

Comment:
Would be good to catch the case where the input format is incorrect (Normally ArrayOutOfBounds if they modified and added extra details.